src/server/types.ts

```diff
@@ -17,4 +17,18 @@ declare namespace ts.server {
         trace?(s: string): void;
         require?(initialPath: string, moduleName: string): RequireResult;
     }
+
+    export type FileServerHost = Pick<
+        ServerHost,
+        | "readFile" | "writeFile" | "fileExists" | "directoryExists" | "getFileSize" | "getModifiedTime"
+        | "getDirectories" | "getCurrentDirectory" | "getExecutingFilePath" | "realpath" | "resolvePath"
+        | "createDirectory" | "setModifiedTime" | "readDirectory" | "watchFile" | "watchDirectory"
+        | "useCaseSensitiveFileNames"> & {
+            deleteFile(path: string, deleteEmptyParentFolders?: boolean): void;
+            ensureFileOrFolder(
+                fileOrDirectoryOrSymLink: { path: string } & ({ content: string, fileSize?: number } | {} | { symLink: string }),
+                ignoreWatchInvokedWithTriggerAsFileCreate?: boolean,

```

Comment:
ignoreWatchInvokedWithTriggerAsFileCreate and ignoreParentWatch should be internal